languages/csharp/exercises/concept/resource-lifetime/ResourceLifetimeTests.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using Xunit;
+
+public class ResourceLifetimeTests
+{
+    enum Exception
+    {
+        InvalidOperationExceptionThrown,
+        NoInvalidOperationExceptionThrown
+    }
+    [Fact]
+    public void Write_good()
+    {
+        var db = new Database();
+        var orm = new Orm(db);
+        orm.Write("good write");
+        Assert.Equal((Database.State.Closed, "good write"),
+            (Database.DbState, Database.lastData));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Write_bad()
+    {
+        Exception result = Exception.NoInvalidOperationExceptionThrown;
+        try
+        {
+            var db = new Database();
+            var orm = new Orm(db);
+            orm.Write("bad write");
+        }
+        catch (InvalidOperationException)
+        {
+            result = Exception.InvalidOperationExceptionThrown;
+        }
+        Assert.Equal((Exception.InvalidOperationExceptionThrown, Database.State.Closed, "bad write"),
+            (result, Database.DbState, Database.lastData));
+    }
```

Comment:
I'm not a big fan of using tuples for the result, but I assume you partially did that because I told you to use one assertion :)

As a future possible improvement, we could consider splitting this method into separate tests, which would allow use to use the `Assert.Throws` method:

```csharp
public void Write_bad_throws_invalidoperationexception()
{
    var db = new Database();
    var orm = new Orm(db);
    Assert.Throws<InvalidOperationException>(() => orm.Write("bad write"));
}
```

This will also result in a nicer assertion error message if it fails. But this is something we can do later.